ports/mimxrt/mphalport.h

```diff
@@ -29,13 +29,22 @@
 
 #include <stdint.h>
 #include "ticks.h"
+#include "pin.h"
 
+#define mp_hal_pin_obj_t const machine_pin_obj_t *
 #define mp_hal_pin_high(p) (GPIO_PinWrite(p->gpio, p->pin, 1U))
 #define mp_hal_pin_low(p) (GPIO_PinWrite(p->gpio, p->pin, 0U))
+#define mp_hal_pin_write(p, value) (GPIO_PinWrite(p->gpio, p->pin, value))
 #define mp_hal_pin_toggle(p) (GPIO_PortToggle(p->gpio, (1 << p->pin)))
+#define mp_hal_pin_read(p) (GPIO_PinRead(p->gpio, p->pin))
 
 void mp_hal_set_interrupt_char(int c);
 
+static inline uint64_t  mp_hal_time_ns(void) {
```

Comment:
is this function needed for this PR to work?

Reply:
:thinking: I just tried compiling without it and it works. Interesting. I could swear that the build was failing without it - that is why I included it.
Ok - will be removed :+1:

Reply:
Yeah, could be. But we (myself and @robert-hh) made sure that both changes are completely stand-alone to make reviewing easier for you. Therefore there is no real need to keep it in this change.